src/runtime/disconnected-callback.ts

```diff
@@ -17,11 +25,12 @@ export const disconnectedCallback = (elm: d.HostElement) => {
       }
     }
 
-    if (BUILD.lazyLoad && BUILD.disconnectedCallback) {
-      safeCall(instance, 'disconnectedCallback');
-    }
-    if (BUILD.cmpDidUnload) {
-      safeCall(instance, 'componentDidUnload');
+    if (!BUILD.lazyLoad) {
+      disconnectInstance(elm);
+    } else if (hostRef.$lazyInstance$) {
+      disconnectInstance(hostRef.$lazyInstance$);
+    } else if (hostRef.$onReadyPromise$) {
+      hostRef.$onReadyPromise$.then(() => disconnectInstance(hostRef.$lazyInstance$));
```

Comment:
Just a small change to resolve a SNC:
```suggestion
    } else if (hostRef?.$lazyInstance$) {
      disconnectInstance(hostRef.$lazyInstance$);
    } else if (hostRef?.$onReadyPromise$) {
      hostRef.$onReadyPromise$.then(() => disconnectInstance(hostRef.$lazyInstance$));
```